frontend/pages/admin/IntegrationsPage/cards/CertificateAuthorities/components/HydrantForm/helpers.ts

```diff
@@ -0,0 +1,149 @@
+import { ICertificatesIntegrationHydrant } from "interfaces/integration";
+
+import valid_url from "components/forms/validators/valid_url";
+
+import { IHydrantFormData } from "./HydrantForm";
+
+// TODO: create a validator abstraction for this and the other form validation files
+
+export interface IHydrantFormValidation {
+  isValid: boolean;
+  name?: { isValid: boolean; message?: string };
+  url?: { isValid: boolean; message?: string };
+  clientId?: {
+    isValid: boolean;
+    message?: string;
+  };
+  clientSecret?: { isValid: boolean; message?: string };
+}
+
+type IMessageFunc = (formData: IHydrantFormData) => string;
+type IValidationMessage = string | IMessageFunc;
+type IFormValidationKey = keyof Omit<IHydrantFormData, "isValid">;
+
+interface IValidation {
+  name: string;
+  isValid: (formData: IHydrantFormData) => boolean;
+  message?: IValidationMessage;
+}
+
+type IFormValidations = Record<
+  IFormValidationKey,
+  { validations: IValidation[] }
+>;
+
+export const generateFormValidations = (
+  hydrantIntegrations: ICertificatesIntegrationHydrant[],
+  isEditing: boolean
+) => {
+  const FORM_VALIDATIONS: IFormValidations = {
+    name: {
+      validations: [
+        {
+          name: "required",
+          isValid: (formData: IHydrantFormData) => {
+            return formData.name.length > 0;
+          },
+        },
+        {
+          name: "invalidCharacters",
```

Comment:
Just saw this while implementing this pattern in one of my PRs -- we should standardize on either positive or negative names for the validations (i.e. either `validCharacters` or `invalidCharacters`). You have both in this one.

Reply:
makes sense, will make this change to positive names in the mew PR